src/app/containers/ImageWithPlaceholder/index.jsx

```diff
@@ -38,10 +38,16 @@ const ImageWithPlaceholder = ({
 }) => {
   const { isAmp } = useContext(RequestContext);
   const [isLoaded, setIsLoaded] = useState(false);
+  const [syncDecoding, setSyncDecoding] = useState(false);
   const imageProps = { alt, src, sizes, width, srcset, fade };
   const imgType = src.split('.').pop();
   const imageToRender = (
-    <Image onLoad={() => setIsLoaded(true)} {...imageProps} />
+    <Image
+      onLoad={() => setIsLoaded(true)}
+      onError={() => setSyncDecoding(true)}
```

Comment:
This looks a great approach but just wondering if we could scope this a bit more? Is there any message we can expect in the onError callback when it doesn't load due to decoding issues: https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror

Just concious this change will cause all images that fail to load to try and load again, regardless of the cause

Reply:
Just to add on to this I think we can use [`EncodingError`](https://heycam.github.io/webidl/#encodingerror) to scope this more

Reply:
Great observation @andrewscfc. Let me explore the error that is returned. I am sure there is a way to scope it down to just this particular use-case.

Reply:
@andrewscfc It looks like the [`onError`](url) event handler API does not say what kind of error occurred at the moment. All we have privy to is that something went wrong when loading this image. Might I be missing something?

@HarveyPeachey it seems we can only know it was an [EncodingError](https://heycam.github.io/webidl/#encodingerror) when we decode the image manually. Here's [one such example](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decode#examples). Should we explore decoding and rendering the image ourselves?

Reply:
I was thinking something like `onError={(event) => event === "decodingError" ? setSyncDecoding(true) : null}`
I don't know exactly what the onError callback receives, this _looks like_ the correct docs: https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror#element.onerror

Might be worth console logging the arguments you get if any?

If you can't scope it, I'd potentially just go with your current solution, it would just mean images attempt to reload once if they fail which isn't that bad I don't think.

Reply:
I tried `console.log(event)` but the `event` recieved is a [generic event](https://developer.mozilla.org/en-US/docs/Web/API/Event). Not sure why the `onError` callback doesn't  receive an [error event](https://developer.mozilla.org/en-US/docs/Web/API/ErrorEvent) instead - it would be much more informative for us.

Reply:
What value do you get for [event.type](https://developer.mozilla.org/en-US/docs/Web/API/Event/type)? Just wondering if something in the object tells you nature of the event.

Anyway, if we can't scope, more than happy to leave this as it is, it could though do with a unit test actually, so faking an error event and confirming the `decoding` prop is applied correctly.

Reply:
`event.type` is `"error"`. 

Yeah, I am afraid we can't scope it without increasing the complexity

Reply:
`https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror#element.onerror`

Looking at the docs could this potentially create a re-render loop?  As `onError` will fire with every single error that an image may produce, for example if the `img` url provided is wrong that will cause a render changing decoding to `sync`, then `onError` will fire again and call `setSyncDecoding(true)`. Does react look at that state hook and prevent a re-render because it is already set to `true`, if it does then we are good and I'm happy with this approach.

Reply:
You could check some of those scenarios in tests @HarveyPeachey, I suspect because the state is stable you _probably_ won't get a render loop.

Reply:
Good point @HarveyPeachey but I believe @andrewscfc is right. The component will re-render on the first failure when our `syncDecoding` state changes from `false` to `true`. The second `onError` event will not trigger re-rendering because state will already be the same as before (`true`).

Reply:
Unit test is ready @andrewscfc and @HarryVerhoef